plugins/ci/pkg/ci/ci.go

```diff
@@ -549,20 +554,33 @@ func readConfigurationFromReader(configHandler io.Reader) (*models.Configuration
 }
 
 func (ci *CIScan) ProcessRepository() ([]*models.ReportInfo, error) {
+	var scanErrorsReportProperties models.ScanErrorsReportProperties // errors encountered during scan
+
 	err := ci.ProcessHelmTemplates()
 	if err != nil {
-		return nil, fmt.Errorf("Error while processing helm templates: %v", err)
+		scanErrorsReportProperties.AddScanErrorsReportResultFromError(err)
 	}
 
 	err = ci.CopyYaml()
 	if err != nil {
-		return nil, fmt.Errorf("Error while copying YAML files: %v", err)
+		scanErrorsReportProperties.AddScanErrorsReportResultFromError(models.ScanErrorsReportResult{
+			ErrorMessage: err.Error(),
+			ErrorContext: "copying yaml files to configuration directory",
+			Kind:         "InternalOperation",
```

Comment:
I specify the required `Kind` field as `InternalOperation` often, because a lot of errors don't map back to a real Kube resource.